packages/terminal-external/src/electron-browser/terminal-external-preference.ts

```diff
@@ -0,0 +1,101 @@
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, interfaces, postConstruct } from 'inversify';
+import {
+    createPreferenceProxy,
+    PreferenceSchema,
+    PreferenceService,
+    PreferenceProxy
+} from '@theia/core/lib/browser';
+import { PreferenceSchemaProvider } from '@theia/core/lib/browser/preferences/preference-contribution';
+import { isWindows, isOSX } from '@theia/core/lib/common/os';
+import { TerminalExternalService, TerminalExternalConfiguration } from '../common/terminal-external';
+
+export const TerminalExternalPreferences = Symbol('TerminalExternalPreferences');
+export type TerminalExternalPreferences = PreferenceProxy<TerminalExternalConfiguration>;
+
+export const TerminalExternalSchemaPromise = Symbol('TerminalExternalSchemaPromise');
+export type TerminalExternalSchemaPromise = Promise<PreferenceSchema>;
+
+export function bindTerminalExternalPreferences(bind: interfaces.Bind): void {
+    bind(TerminalExternalSchemaPromise).toDynamicValue(
+        ctx => getTerminalExternalSchema(ctx.container.get(TerminalExternalService))
+    ).inSingletonScope();
+    bind(TerminalExternalPreferences).toDynamicValue(
+        ctx => createPreferenceProxy(
+            ctx.container.get(PreferenceService),
+            ctx.container.get(TerminalExternalSchemaPromise),
+        )
+    ).inSingletonScope();
+    bind(TerminalExternalPreferenceService).toSelf().inSingletonScope();
+}
+
+@injectable()
+export class TerminalExternalPreferenceService {
+
+    @inject(TerminalExternalPreferences)
+    private readonly preferences: TerminalExternalPreferences;
+
+    @inject(PreferenceSchemaProvider)
+    private readonly preferenceSchemaProvider: PreferenceSchemaProvider;
+
+    @inject(TerminalExternalSchemaPromise)
+    private readonly promisedSchema: TerminalExternalSchemaPromise;
+
+    @postConstruct()
+    protected setPreferenceSchema(): void {
+        this.promisedSchema.then(schema => this.preferenceSchemaProvider.setSchema(schema));
+    }
+
+    /**
+     * Get the external terminal configurations from preferences.
+     */
+    getTerminalExternalConfiguration(): TerminalExternalConfiguration {
+        return {
+            'terminal.external.linuxExec': this.preferences['terminal.external.linuxExec'],
+            'terminal.external.osxExec': this.preferences['terminal.external.osxExec'],
+            'terminal.external.windowsExec': this.preferences['terminal.external.windowsExec'],
+        };
+    }
+}
```

Comment:
Do we still need that since we have a preference proxy now?

Reply:
@marechal-p 
Just want to clarify a couple of things:
+ `this.preferences` here refers to the `TerminalExternalPreferences` (https://github.com/eclipse-theia/theia/pull/9186/files#diff-19f28c984d3bca321f863492f25995d64fb47fc712eed3b05c8b1a91ac46f0f8R50-R51), not the `PreferenceService`.
+ This method is used to get the preferences (front-end) and pass them to `openTerminal()` from the service (back-end) so I think it's needed unless you meant we should get the preferences directly in the back-end via proxy?

Let me know if I got your question wrong!

Reply:
OK makes sense, I just thought this object could be created trivially without requiring it's own service like that, but it is fine.